stan/math/prim/scal/fun/is_nan.hpp

```diff
@@ -14,7 +14,10 @@ namespace math {
  * @param x Value to test.
  * @return <code>true</code> if the value is NaN.
  */
-inline bool is_nan(double x) { return std::isnan(x); }
+template <typename T>
+inline bool is_nan(T x) {
```

Comment:
Hmm... this looks odd. In `develop`, we have 3 definitions of `is_nan`: prim/scal, rev/scal, fwd/scal. They are implemented using function overloading.

I think you either want to go the same route or you want to make this function the template function (as you have it) and then change the implementations in rev and fwd to be template specializations.

Reply:
Does that make sense? If it's not clear what I'm talking about, we can discuss.